src/hooks/PushNotification.ts

```diff
@@ -0,0 +1,67 @@
+import * as Notifications from 'expo-notifications';
+import { AppState, AppStateStatus } from 'react-native';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { Subscription } from '@unimodules/react-native-adapter';
+import { readFromStore } from '../helpers';
+import { initializePushPermissions, PushNotificationStorageKeys, updatePushToken } from '../pushNotifications';
+
+type NotificationHandler = (arg: Notifications.Notification) => void;
+type ResponseHandler = (arg: Notifications.NotificationResponse) => void;
+
+export const usePushNotifications = (
+  notificationHandler?: NotificationHandler,
+  interactionHandler?: ResponseHandler,
+  behavior?: Notifications.NotificationBehavior
+) : void => {
+
+  const notificationListener = useRef<Subscription>();
+  const responseListener = useRef<Subscription>();
+
+  const [currentAppState, setCurrentAppState] = useState<AppStateStatus>();
+
+  const onGetActive = useCallback(async (nextState: AppStateStatus) => {
+    if (currentAppState != nextState) {
+      setCurrentAppState(nextState);
+      const inAppPermission = await readFromStore(PushNotificationStorageKeys.IN_APP_PERMISSION);
+      if (nextState === 'active') {
+        inAppPermission && updatePushToken();
+      }
+    }
+  }, []); // empty dependencies because it will only used once in the "mountEffect" below
+
+  useEffect(() => {
+    initializePushPermissions();
+
+    AppState.addEventListener('change', onGetActive);
+
+    // This listener is fired whenever a notification is received while the app is foregrounded
+    notificationListener.current
+      = notificationHandler && Notifications.addNotificationReceivedListener((notification) => {
```

Comment:
haven't seen the `=` in the beginning of a line yet.
does eslint not complain about this?
i think this file needs to be re-formatted in a few places.

Reply:
no eslint does not complain here. same with any other operator. you can have

```ts
conditionA &&
conditionB
```
as well as 
```ts
conditionA
&& conditionB
```

I changed it to have the assignment operator in the end of the previous line in e2f9b849f1157cec42932f4fc81cf4ce13f20fc8

Reply:
thanks. i will have a look at eslint regarding this.